pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -86,6 +96,30 @@ func isVolumeConflict(volume *v1.Volume, pod *v1.Pod) bool {
 	return false
 }
 
+func (pl *VolumeRestrictions) isAccessModeConflict(volume *v1.Volume, pod *v1.Pod) (bool, framework.Code, string) {
+	for _, existingVolume := range pod.Spec.Volumes {
+		if volume.PersistentVolumeClaim != nil && existingVolume.PersistentVolumeClaim != nil {
+			claimName, existingClaimName := volume.PersistentVolumeClaim.ClaimName, existingVolume.PersistentVolumeClaim.ClaimName
+			if claimName != existingClaimName {
+				continue
+			}
+
+			existingPVC, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(existingClaimName)
+			if err != nil {
+				// If there are errors fetching the PVC, then try again in another scheduling cycle.
+				return true, framework.Error, ErrReasonUnexpected
+			}
+
+			// Enforce ReadWriteOncePod access mode. This is also enforced during volume mount in kubelet.
+			if utilfeature.DefaultFeatureGate.Enabled(features.ReadWriteOncePod) && volumeutil.AccessModesContains(existingPVC.Spec.AccessModes, v1.ReadWriteOncePod) {
+				return true, framework.UnschedulableAndUnresolvable, ErrReasonReadWriteOncePodConflict
+			}
```

Comment:
Yeah this makes sense. I implemented it like so because it seemed like if the access modes contain `ReadWriteOncePod`, then there is the potential for a pod to use it in this fashion. If that is the case, we want to make sure that access mode is supported (e.g. kubelet will enforce this behavior on mount).

Reply:
@jsafrane I synced with @msau42 and it was suggested we check this during validation, since use of `ReadWriteOncePod` renders any other included access modes useless. Please see [this commit](https://github.com/kubernetes/kubernetes/pull/102028/commits/6eb291ff81a51f07ca254469d8bcfa16d4ff0635) for the fix. Thoughts on this approach?

Reply:
As I wrote above, `AccessMode` semantics is not very well defined, esp. when it's an array. Requiring `RWOP` to be the only item there works for me. I commented below, a note about it would be nice. And then we'll see how far it gets during API review.

Reply:
@jsafrane did you mean scheduling or API validation? I don't think the scheduler should verify this.